libs/blocks/global-navigation/global-navigation.js

```diff
@@ -949,8 +949,12 @@ class Gnav {
 
 export default async function init(block) {
   try {
-    const { locale, mep } = getConfig();
-    const url = getMetadata('gnav-source') || `${locale.contentRoot}/gnav`;
+    const { locale, mep, dynamicNavKey } = getConfig();
+    let url = getMetadata('gnav-source') || `${locale.contentRoot}/gnav`;
+    if (dynamicNavKey) {
+      const { default: dynamicNav } = await import('../../features/dynamic-navigation.js');
+      url = dynamicNav(url, dynamicNavKey);
+    }
```

Comment:
I think we should probably have a new function `getGnavSource` (or something along those lines) that produces the url and contains all the logic for generating the url within it.

Reply:
I agree with Raghav, we should extract it into a separate function that generates the URL.

Reply:
Hmm. I am not so sure that I agree.  The `dynamic nav` is an opt in feature requiring a consumer to update their `config` to have the `dynamicNavKey` in order to kick off the experience. 

Though it may make the `init` function a bit cleaner, a function combining the two would mix both the default behavior and an opt in feature, which I am not certain aids in readability. 

Interested to see what you think.

Reply:
In the future we may have logic added or subtracted from our hypothetical `getSource` function. `init` is and should be agnostic about the details of getting the url and that is communicated well by a separate function.

The separation between default and opt in behaviour can be communicated well enough in the function itself by the if statement you'll have to add. And the notion of what constitutes default vs non default behavior now and in the future is malleable enough that it makes sense to cordon it off from our main `init` logic so that it's clear in the future where gnav url generation logic should go.

Reply:
I appreciate your thoughtful and full response! I see what you're saying. To be honest, I am still on the fence because it feels a bit like premature optimization and we'd be adding 3 lines of code by pulling these 5 out into a function.

Maybe @mokimo can weigh in since he recommended the approach.

Reply:
@sharmrj So you'd think something like this?

```js
const getSource = async () => {
    const { locale, dynamicNavKey } = getConfig();
    let url = getMetadata('gnav-source') || `${locale.contentRoot}/gnav`;
    if (dynamicNavKey) {
      const { default: dynamicNav } = await import('../../features/dynamic-navigation.js');
      url = dynamicNav(url, dynamicNavKey);
    }
}

export default async function init(block) {
  try {
    const { locale, mep } = getConfig();
    const url = await getSource()
    const content = await fetchAndProcessPlainHtml({ url })
      .catch((e) => lanaLog({
        message: `Error fetching gnav content url: ${url}`,
        e,
        tags: 'errorType=error,module=gnav',
      }));
....
```

@sharmrj and @bandana147 are the new gnav code owners 😁 So if that's how they'd prefer it, no objections

Reply:
Yup. That's what I was envisioning.

Reply:
Sounds good. I'll make those changes you requested @sharmrj, and I'll be sure to update the unit tests to reflect the new function. Hopefully should have something for you to look at a bit later today. I'll be sure to update the docs as well @mokimo. 

Thank you both for your input!